src/managers/common/nativePythonFinder.ts

```diff
@@ -380,6 +384,223 @@ function getPythonSettingAndUntildify<T>(name: string, scope?: Uri): T | undefin
+            }
+            // If path doesn't exist, add it anyway as it might exist in the future
+            // This handles cases where:
+            // - Virtual environments might be created later
+            // - Network drives that might not be mounted yet
+            // - Symlinks to directories that might be created
+            else {
+                traceLog('Path does not exist currently, adding for future resolution when environments may be created:', trimmedPath);
+                searchDirectories.push(trimmedPath);
+            }
+        } catch (error) {
+            traceLog('Error processing search path:', searchPath, 'Error:', error);
+        }
+    }
+
+    // Remove duplicates and return
+    const uniquePaths = Array.from(new Set(searchDirectories));
+    traceLog('getAllExtraSearchPaths completed. Total unique search directories:', uniquePaths.length, 'Paths:', uniquePaths);
+    return uniquePaths;
+}
+
+/**
+ * Gets searchPaths setting value using proper VS Code settings precedence.
+ * Checks workspaceFolder, then workspace, then user level settings.
+ * @returns Array of search paths from the most specific scope available
+ */
+function getSearchPathsWithPrecedence(): string[] {
+    try {
+        // Use VS Code configuration inspection to handle precedence automatically
+        const config = getConfiguration('python-env');
+        const inspection = config.inspect<string[]>('searchPaths');
+        
+        // VS Code automatically handles precedence: workspaceFolder -> workspace -> user
+        // We check each level in order and return the first one found
+        if (inspection?.workspaceFolderValue) {
+            traceLog('Using workspaceFolder level searchPaths setting');
+            return untildifyArray(inspection.workspaceFolderValue);
+        }
+        
+        if (inspection?.workspaceValue) {
+            traceLog('Using workspace level searchPaths setting');
+            return untildifyArray(inspection.workspaceValue);
+        }
+        
+        if (inspection?.globalValue) {
+            traceLog('Using user level searchPaths setting');
+            return untildifyArray(inspection.globalValue);
+        }
+        
+        // Default empty array
+        traceLog('No searchPaths setting found at any level, using empty array');
+        return [];
+    } catch (error) {
+        traceLog('Error getting searchPaths with precedence:', error);
+        return [];
+    }
+}
+
+/**
+ * Applies untildify to an array of paths
+ * @param paths Array of potentially tilde-containing paths
+ * @returns Array of expanded paths
+ */
+function untildifyArray(paths: string[]): string[] {
+    return paths.map(p => untildify(p));
+}
+
+/**
+ * Handles migration from python.venvFolders to python-env.searchPaths.
+ * Only migrates if python.venvFolders exists and searchPaths is different.
+ */
+async function handleVenvFoldersMigration(): Promise<void> {
```

Comment:
there is another legacy setting "python.venvFolders". Add migration support for this one as well